[mypy]
python_version = 3.10
incremental = True
plugins = pydantic.mypy
check_untyped_defs = True
warn_unreachable = True
# TODO: Re-enable namespace_packages? Is enabled by default in versions >=0.990
#       and triggers some new errors.
#    https://mypy-lang.blogspot.com/2022/11/mypy-0990-released.html
namespace_packages = False

[mypy-cftime.*]
ignore_missing_imports = True

[mypy-funcy.*]
# Funcy has no plans for type annotation:
#    https://github.com/Suor/funcy/issues/53#issuecomment-688232443
ignore_missing_imports = True

[mypy-pytest.*]
ignore_missing_imports = True

[mypy-luigi.*]
ignore_missing_imports = True

[mypy-PyQt5.*]
ignore_missing_imports = True

[mypy-rarfile.*]
ignore_missing_imports = True

[mypy-yamale.*]
ignore_missing_imports = True

[mypy-fiona.*]
ignore_missing_imports = True

[mypy-shapely.*]
ignore_missing_imports = True

[mypy-numpy.*]
ignore_missing_imports = True

[mypy-pandas.*]
ignore_missing_imports = True

[mypy-geopandas.*]
ignore_missing_imports = True

[mypy-osgeo.*]
ignore_missing_imports = True

[mypy-pyproj.*]
ignore_missing_imports = True

[mypy-rasterio.*]
ignore_missing_imports = True

[mypy-netCDF4.*]
ignore_missing_imports = True

[mypy-qgis.*]
ignore_missing_imports = True

[mypy-humanize.*]
ignore_missing_imports = True

[mypy-anytree.*]
ignore_missing_imports = True
